pkg/agent/secondarynetwork/podwatch/controller_test.go

```diff
@@ -936,3 +940,143 @@ func testPodControllerStart(ctrl *gomock.Controller) (
+
+	ovsPort3 := ovsconfig.OVSPortData{
+		UUID: uuid3, Name: "p3", OFPort: -1,
+		ExternalIDs: convertExternalIDMap(cniserver.BuildOVSPortExternalIDs(
+			interfacestore.NewContainerInterface("p3", uuid3, "pod3", "ns3", "eth0", p2NetMAC, []net.IP{p2NetIP}, 100)))}
+
+	ovsPort4 := ovsconfig.OVSPortData{
+		UUID:   uuid3,
+		Name:   "unknownIface",
+		OFPort: 20,
+		ExternalIDs: map[string]string{
+			"unknownKey": "unknownValue"}}
+
+	// Interface configurations
+	iface1 := cniserver.ParseOVSPortInterfaceConfig(&ovsPort1, &interfacestore.OVSPortConfig{PortUUID: ovsPort1.UUID, OFPort: ovsPort1.OFPort})
+	iface2 := cniserver.ParseOVSPortInterfaceConfig(&ovsPort2, &interfacestore.OVSPortConfig{PortUUID: ovsPort2.UUID, OFPort: ovsPort2.OFPort})
+	iface3 := cniserver.ParseOVSPortInterfaceConfig(&ovsPort3, &interfacestore.OVSPortConfig{PortUUID: ovsPort3.UUID, OFPort: ovsPort3.OFPort})
+
+	return map[string]string{"uuid1": uuid1, "uuid2": uuid2, "uuid3": uuid3}, []ovsconfig.OVSPortData{ovsPort1, ovsPort2, ovsPort3, ovsPort4}, []*interfacestore.InterfaceConfig{iface1, iface2, iface3}
+}
+
+func setupMockController(t *testing.T) (*gomock.Controller, *ovsconfigtest.MockOVSBridgeClient, *podwatchtesting.MockInterfaceConfigurator, *podwatchtesting.MockIPAMAllocator, *PodController) {
+	ctrl := gomock.NewController(t)
+	mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(ctrl)
+	interfaceConfigurator := podwatchtesting.NewMockInterfaceConfigurator(ctrl)
+	mockIPAM := podwatchtesting.NewMockIPAMAllocator(ctrl)
+
+	store := interfacestore.NewInterfaceStore()
+	pc := &PodController{
+		ovsBridgeClient:       mockOVSBridgeClient,
+		interfaceStore:        store,
+		cniCache:              sync.Map{},
+		interfaceConfigurator: interfaceConfigurator,
+		ipamAllocator:         mockIPAM,
+	}
+	return ctrl, mockOVSBridgeClient, interfaceConfigurator, mockIPAM, pc
+
+}
+
+func TestInitializeSecondaryInterfaceStore(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	// Test Case 1: OVSBridgeClient is nil
+	store := interfacestore.NewInterfaceStore()
+	pc := &PodController{
+		ovsBridgeClient: nil,
+		interfaceStore:  store,
+	}
+	err := pc.initializeSecondaryInterfaceStore()
+	assert.NoError(t, err, "No error when OVSBridgeClient is nil")
+
+	// Test Case 2: OVSBridgeClient returns an error
+	ctrl, mockOVSBridgeClient, _, _, pc := setupMockController(t)
+	defer ctrl.Finish()
+
+	mockOVSBridgeClient.EXPECT().GetPortList().Return(nil, ovsconfig.NewTransactionError(fmt.Errorf("Failed to list OVS ports"), true))
+	err = pc.initializeSecondaryInterfaceStore()
+	assert.Error(t, err, "Failed to list OVS ports")
+
+	// Test Case 3: OVSBridgeClient returns valid ports
+	uuids, ovsPorts, _ := createTestInterfaces()
+	mockOVSBridgeClient.EXPECT().GetPortList().Return(ovsPorts, nil)
+
+	err = pc.initializeSecondaryInterfaceStore()
+	assert.NoError(t, err, "OVS ports list successfully")
```

Comment:
you need to use `require` in this case, as the rest of the test is guaranteed to fail or even panic

please review other assertions as well